### PR TITLE
update tests for `dpnp.fft` module to avoid failures and warning with NumPy 2.0

### PR DESCRIPTION
**Pull request by vtavana**
In this PR, a few tests are updated to avoid failure in comparison with NumPy 2.0. Also, third party tests are updated to avoid deprecated warning from NumPy 2.0 for "if `s` is given, `axes` cannot be `None`".

- [x] Have you provided a meaningful PR description?
- [x] Have you added a test, reproducer or referred to issue with a reproducer?
- [x] Have you tested your changes locally for CPU and GPU devices?
- [x] Have you made sure that new changes do not introduce compiler warnings?
- [ ] Have you checked performance impact of proposed changes?
- [ ] If this PR is a work in progress, are you filing the PR as a draft?

### CODE REVIEW

File: tests/test_sycl_queue.py [hunk 1]
```diff
@@ -1283,111 +1283,108 @@ def test_out_multi_dot(device):
...
-    assert_dtype_allclose(result, expected, check_only_type_kind=True)
+        expected = getattr(numpy.fft, func)(data)
+        result = getattr(dpnp.fft, func)(dpnp_data)
+        assert_dtype_allclose(result, expected, factor=24)
 
-    expected_queue = dpnp_data.sycl_queue
-    result_queue = result.sycl_queue
-    assert_sycl_queue_equal(result_queue, expected_queue)
+        expected_queue = dpnp_data.get_array().sycl_queue
+        result_queue = result.get_array().sycl_queue
```
Comment (antonwolfy) [inline tests/test_sycl_queue.py:1305]: dpnp.ndarray also has property `sycl_queue`:
```suggestion
        expected_queue = dpnp_data..sycl_queue
        result_queue = result..sycl_queue
```

File: tests/third_party/cupy/fft_tests/test_fft.py [hunk 2]
```diff
@@ -142,8 +142,8 @@ def test_ifft(self, xp, dtype):
                 # {"shape": (0, 5), "s": None, "axes": None}, # mkl_fft gh-110
                 # {"shape": (2, 0, 5), "s": None, "axes": None}, # mkl_fft gh-110
                 # {"shape": (0, 0, 5), "s": None, "axes": None}, # mkl_fft gh-110
-                {"shape": (3, 4), "s": (0, 5), "axes": None},
-                {"shape": (3, 4), "s": (1, 0), "axes": None},
+                {"shape": (3, 4), "s": (0, 5), "axes": (0, 1)},
```
Comment (antonwolfy) [inline tests/third_party/cupy/fft_tests/test_fft.py:145]: can we add a comment, like:
```suggestion
                # if s is given, axes can not be None
                {"shape": (3, 4), "s": (0, 5), "axes": (0, 1)},
```

File: tests/test_fft.py [hunk 3]
```diff
@@ -69,15 +69,33 @@ def test_fft_ndim(self, dtype, shape, norm):
     @pytest.mark.parametrize("norm", [None, "forward", "ortho"])
     def test_fft_1D(self, dtype, n, norm):
         x = dpnp.linspace(-1, 1, 11, dtype=dtype)
-        a = dpnp.sin(x)
+        a = dpnp.sin(x)  # a.dtype is float16 if a.dtype is bool
         a_np = dpnp.asnumpy(a)
 
+        factor = 140 if dtype == dpnp.bool else 8
```
Comment (antonwolfy) [inline tests/test_fft.py:75]: Do we need to exclude `bool` dtype for that test if we have a dedicated test below?

**Review by antonwolfy — APPROVED** [review]
FFT tests are passing now with NumPy 2.0. Thank you @vtavana for resolving that

File: tests/test_fft.py (hunk 3 above)
Comment (vtavana) [reply tests/test_fft.py:75]: In this test when `dtype` is `bool`, `x.dtype` is bool but `a.dtype` would be `float16` which is not included by default. So, I guess it worth to keep it as it is. I added the additional test to cover `bool` dtype as well.

File: tests/third_party/cupy/fft_tests/test_fft.py (hunk 2 above)
Comment (vtavana) [reply tests/third_party/cupy/fft_tests/test_fft.py:145]: I have already added a comment that some of the tests are updated. 
https://github.com/IntelPython/dpnp/blob/6cd9389804839edb825152e30248eab9b9921dd6/tests/third_party/cupy/fft_tests/test_fft.py#L123-L125

I do not think we need to indicate it individually for each test. Besides, I believe sooner or later `cupy` will update the tests.

Comment (antonwolfy) [reply tests/third_party/cupy/fft_tests/test_fft.py:145]: I see, I missed that. Then I agree there is no reason to duplicate with extra comment few lines later.

**Review by antonwolfy — APPROVED** [review]

**Pull request merged**